### PR TITLE
wrap a function as a jet operator

### PR DESCRIPTION
**Pull request by ziyiyin97**
This PR provides the functionality to wrap a generic Julia function as a Jet operator. The forward and adjoint actions of the Jacobian are provided by Julia's AD system.

### CODE REVIEW

**Comment by samtkaplan** [comment]
This is great.  Can we use https://pkgdocs.julialang.org/v1/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions) rather than Requires, please?

File: Project.toml [hunk 1]
```diff
@@ -4,15 +4,24 @@ version = "1.1.3"
...
 [compat]
 FFTW = "1"
+Flux = "0.12, 0.13"
 Jets = "^1.2"
-Requires = "1"
 SpecialFunctions = "1, 2"
 julia = "1"
+
+[extras]
+Flux = "587475ba-b771-5e3f-ad9e-33799f191a9c"
```
Comment (ziyiyin97) [inline Project.toml:27]: This makes sure that it also works for older Julia versions https://pkgdocs.julialang.org/v1/creating-packages/#Using-an-extension-while-supporting-older-Julia-versions

File: Project.toml [hunk 2]
```diff
@@ -1,16 +1,27 @@
 name = "JetPack"
 uuid = "24ef3835-3876-54c3-8a7a-956cf69ca0b2"
-version = "1.1.2"
+version = "1.1.3"
```
Comment (samtkaplan) [inline Project.toml:3]: I guess this should be 1.2.0 since you are adding a new feature.

File: Project.toml [hunk 3]
```diff
@@ -1,16 +1,27 @@
...
+
+[extensions]
+JetPackFluxExt = "Flux"
+
 [compat]
 FFTW = "1"
+Flux = "0.12, 0.13"
 Jets = "^1.2"
 SpecialFunctions = "1, 2"
 julia = "1"
```
Comment (samtkaplan) [inline Project.toml:24]: should this be Julia 1.9 since the weak deps stuff is only supported since Julia 1.9?

**Review by samtkaplan — APPROVED** [review]
Thanks.  I'm wondering if in the future, we should move this functionality into Jets.jl.  That is, if one creates a Jet operator without df or df' defined, then Jets should always attempt to use auto-diff?

**Comment by ziyiyin97** [comment]
> should this be Julia 1.9 since the weak deps stuff is only supported since Julia 1.9?

https://pkgdocs.julialang.org/v1/creating-packages/#Using-an-extension-while-supporting-older-Julia-versions this says as long as we put the weak deps in extra section, it should work for older versions. However, I haven't tested it on older versions yet. Since there is a CI for Julia 1.6, could you trigger the CI and see how it goes?

> I'm wondering if in the future, we should move this functionality into Jets.jl. That is, if one creates a Jet operator without df or df' defined, then Jets should always attempt to use auto-diff?

This is a very strong assumption IMO ... I personally do not suggest doing that but I will leave that up to you folks.

**Pull request merged**